### ISSUE TITLE
I meet with a bug when writing my code under the framework of IDECF.py

### ISSUE THREAD

**Issue by q21027** [Armanfard-Lab/IDECF#1]
When my code comes to the process similar to FCM_NET_train, the first epoch is normal, but the second one throws a error: Trying to backward through the graph a second time. Why the second epoch throws this kind of error?

**Comment by Armanfard-Lab** [comment]
You must detach variables from FCM_net when training AE using .detach() method
